nf_core/pipelines/download/utils.py

```diff
@@ -81,3 +82,24 @@ def intermediate_file_no_creation(output_path: Path) -> Generator[Path, None, No
     except:
         tmp.cleanup()
         raise
+
+
+@contextlib.contextmanager
+def intermediate_dir_with_cd(original_dir: Path, base_dir: Path = Path(".")):
+    """
+    Context manager to provide and change into a tempdir and ensure its removal and retrun to the
+    original_dir upon exceptions.
+    """
+
+    if not original_dir.is_dir():
+        raise DownloadError(f"Unable to setup temporary dir, original_dir does not exist: {original_dir}")
+
+    tmp = tempfile.TemporaryDirectory(dir=base_dir)
+    try:
+        os.chdir(tmp.name)
+        yield tmp
+        os.chdir(original_dir)
+        tmp.cleanup()
+    except:  # noqa: E722
+        os.chdir(original_dir)
+        tmp.cleanup()
```

Comment:
I don't think the cleanup is needed, because tempfile should take care of that already